yaksh/models.py

```diff
@@ -1474,9 +1477,21 @@ def regrade(self, question_id):
             answer = user_answer.answer
         json_data = question.consolidate_answer_data(answer) \
             if question.type == 'code' else None
-        result = self.validate_answer(answer, question, json_data)
+        result = self.validate_answer(answer, question,
+                                      json_data, user_answer.id)
+        if question.type == "code":
+            check_result = result
+            url = 'http://localhost:%s' % SERVER_POOL_PORT
+            while check_result['status'] != "done":
+                if (check_result['status'] == "running" or
+                    check_result['status'] == "not started" or
+                   check_result['status'] == "unknown"):
+                    check_result = get_result_from_code_server(
```

Comment:
You can use block=True argument, so we do not need to use while
It will wait till the code server returns the result.